public/app/features/alerting/unified/components/rules/central-state-history/CentralAlertHistoryScene.tsx

```diff
@@ -189,35 +217,125 @@ function getSceneQuery(datasource: DataSourceInformation) {
+          .setCustomFieldConfig('lineInterpolation', LineInterpolation.Linear)
+          .setCustomFieldConfig('lineWidth', 1)
+          .setCustomFieldConfig('barAlignment', 0)
+          .setCustomFieldConfig('spanNulls', false)
+          .setCustomFieldConfig('insertNulls', false)
+          .setCustomFieldConfig('showPoints', VisibilityMode.Auto)
+          .setCustomFieldConfig('pointSize', 5)
+          .setCustomFieldConfig('stacking', { mode: StackingMode.None, group: 'A' })
+          .setCustomFieldConfig('gradientMode', GraphGradientMode.Hue)
+          .setCustomFieldConfig('scaleDistribution', { type: ScaleDistribution.Linear })
+          .setOption('legend', { showLegend: false, displayMode: LegendDisplayMode.Hidden })
+          .setOption('tooltip', { mode: TooltipDisplayMode.Single })
+          .setNoValue('No events found')
+          .build() as VizPanel<Options, FieldConfig>,
+    }),
+  });
+}
 
-      .setNoValue('No events found')
-      .build(),
+export class EventsPanelScenesObject extends SceneObjectBase<
+  FilterScenesObjectState & { datasource: DataSourceInformation; body: VizPanel<Options, FieldConfig> }
+> {
+  public static Component = EventsPanelScenesObjectRenderer;
+  private _onAnyVariableChanged(): void {
+    this.forceRender();
+  }
+
+  public _onActivate() {
+    const stateToFilterVariable = sceneGraph.lookupVariable(STATE_FILTER_TO, this);
+    if (stateToFilterVariable instanceof CustomVariable) {
+      this.setState({ ...this.state, valueInStateToFilter: stateToFilterVariable.state.value });
+      this._subs.add(
+        stateToFilterVariable?.subscribeToState((newState, prevState) => {
+          this.setState({ ...prevState, valueInStateToFilter: newState.value });
+          const query = getSceneQuery(this.state.datasource);
+          query.runQueries();
+        })
+      );
+    }
+    const stateFromFilterVariable = sceneGraph.lookupVariable(STATE_FILTER_FROM, this);
+    if (stateFromFilterVariable instanceof CustomVariable) {
+      this.setState({ ...this.state, valueInStateFromFilter: stateFromFilterVariable.state.value });
+      this._subs.add(
+        stateFromFilterVariable?.subscribeToState((newState, prevState) => {
+          this.setState({ ...prevState, valueInStateFromFilter: newState.value });
+          const query = getSceneQuery(this.state.datasource);
+          query.runQueries();
+        })
+      );
+    }
+    const labelsFilterVariable = sceneGraph.lookupVariable(LABELS_FILTER, this);
+    if (labelsFilterVariable instanceof TextBoxVariable) {
+      this.setState({ ...this.state, valueInLabelFilter: labelsFilterVariable.state.value });
+      this._subs.add(
+        labelsFilterVariable?.subscribeToState((newState, prevState) => {
+          this.setState({ ...prevState, valueInLabelFilter: newState.value });
+          const query = getSceneQuery(this.state.datasource);
+          query.runQueries();
+        })
+      );
+    }
+  }
```

Comment:
why all these variable state subscriptions? 

SceneQueryRunner will run queries automatically if they have a variable that changed value